metaserver/src/main/scala/scala/meta/languageserver/Compiler.scala

```diff
@@ -217,4 +204,17 @@ object Compiler extends LazyLogging {
     f(r)
     r
   }
+  def typeOfTree(c: Global)(t: c.Tree): Option[String] = {
```

Comment:
In #51 I moved all logic into `compiler.SignatureHelpProvider`, I think it would make sense to have a corresponding `compiler.HoverProvider` object, even if the method is small.

Reply:
Yes, that's what I intend to do in this PR. I opened it for early feedback, but it's not quite ready for merge :) thanks for the review!